x-pack/plugins/maps/server/data_indexing/get_indexes_matching_pattern.ts

```diff
@@ -20,14 +21,17 @@ export async function getMatchingIndexes(
     const matchingIndexes = indexResults
       .map((indexRecord) => indexRecord.index)
       .filter((indexName) => !!indexName);
-    return {
-      success: true,
-      matchingIndexes: matchingIndexes as string[],
-    };
+    return response.ok({ body: { success: true, matchingIndexes: matchingIndexes as string[] } });
   } catch (error) {
-    return {
-      success: false,
-      error,
-    };
+    const errorStatusCode = error.meta?.statusCode;
+    if (errorStatusCode === 404) {
+      return response.ok({ body: { success: true, matchingIndexes: [] } });
+    } else {
+      logger.error(error);
+      return response.custom({
+        body: { success: false, message: 'Error searching for matching indexes' },
+        statusCode: 500,
```

Comment:
I don't think this should return a 500. there are lots of reasons for the failure other then server error, such as permission problems. I think the response should still be 200 with success as false.

Reply:
I've covered the cases for unauth and forbidden access. Also added handling on the client side for when there really is an issue returning a 500 response

Reply:
Why are you making this change in this PR? Its not part of the bug fix and adds a lot of unneeded complexity to the route. I think the route error handling was fine as is and should not be changed.